shared/oae/api/oae.api.js

```diff
@@ -171,17 +171,120 @@ define(['oae.api.admin', 'oae.api.authentication', 'oae.api.config', 'oae.api.co
+                    }
+                });
+            });
+        };
+
+        /**
+         * Accept the invitation if the query string of the specified location indicates there is an
+         * invitation to be accepted
+         *
+         * @param  {String}         location            The URL whose location to check for invitation info
+         * @param  {Function}       callback            Invoked when the accept invitation request is complete
+         * @param  {Resource[]}     callback.resources  Indicates the resources that were accepted. If the request failed, this will be unspecified
+         */
+        var acceptInvitation = function(location, callback) {
+            var invitationToken = oae.api.util.url(location).param('invitationToken');
+            if (!invitationToken) {
+                return callback();
+            }
+
+            // We need to ensure we can handle activities that happen as a result of accepting this
+            // invitation, so set up the push API a bit earlier than we normally would have
+            oae.api.push.init(function() {
+
+                // Accept the invitation
+                oae.api.user.acceptInvitation(invitationToken, function(err, result) {
+                    if (err && err.code !== 404) {
+                        oae.api.util.notification(
+                            oae.api.i18n.translate('__MSG__EMAIL_INVITATION_FAILED__'),
+                            oae.api.i18n.translate('__MSG__AN_ERROR_OCCURRED_WHILE_ACCEPTING_YOUR_INVITATION__'),
+                            'error');
+                        return callback();
+                    } else if (err) {
+                        return callback();
+                    }
+
+                    // Accepting an invitation will set a verified email if there wasn't one already,
+                    // so do that here to avoid giving the user a pop-up
+                    oae.data.me.email = oae.data.me.email || result.email;
 
-            // Show the user details widget if a valid name or email address need to be provided
-            if (needsToProvideDisplayName || needsToProvideEmail) {
-                oae.api.widget.insertWidget('userdetails');
+                    return callback(result.resources);
+                });
+            });
+        };
 
-            // Show the Terms and Conditions widget if the user needs to accept the Terms and Conditions
-            } else if (oae.data.me.needsToAcceptTC) {
-                oae.api.widget.insertWidget('termsandconditions', null, null, true);
+        /**
+         * Verify the user's email if there is instruction in the query string to do so with the
+         * specified verification token. If there is no `verifyEmail` query string parameter, then
+         * this will simply return doing no work.
+         *
+         * @param  {Function}   callback    Invoked when the email verification has completed, regardless if it was successful or failed
```

Comment:
Similarly, the function handles the errors and the consumer should not try to do anything with any error, so it's not our standard callback function.